test/test_utils/__init__.py

```diff
@@ -2370,3 +2392,95 @@ def get_image_spec_from_buildspec(image_uri, dlc_folder_path):
         raise ValueError(f"No corresponding entry found for {image_uri} in {buildspec_path}")
 
     return matched_image_spec
+
+
+def get_instance_type_base_dlami(instance_type, region, linux_dist="UBUNTU_20"):
+    """
+    Get Instance types based on EC2 instance, see https://docs.aws.amazon.com/dlami/latest/devguide/important-changes.html
+    OSS Nvidia Driver DLAMI supports the following: ["g4dn.xlarge",
+                                                     "g4dn.2xlarge",
+                                                     "g4dn.4xlarge",
+                                                     "g4dn.8xlarge",
+                                                     "g4dn.16xlarge",
+                                                     "g4dn.12xlarge",
+                                                     "g4dn.metal",
+                                                     "g4dn.xlarge",
+                                                     "g5.xlarge",
+                                                     "g5.2xlarge",
+                                                     "g5.4xlarge",
+                                                     "g5.8xlarge",
+                                                     "g5.16xlarge",
+                                                     "g5.12xlarge",
+                                                     "g5.24xlarge",
+                                                     "g5.48xlarge",
+                                                     "p4d.24xlarge",
+                                                     "p4de.24xlarge",
+                                                     "p5.48xlarge",]
+
+    Proprietary Nvidia Driver DLAMI supports the following: ["p3.2xlarge",
+                                                             "p3.8xlarge",
+                                                             "p3.16xlarge",
+                                                             "p3dn.24xlarge",
+                                                             "g3s.xlarge",
+                                                             "g3.4xlarge",
+                                                             "g3.8xlarge",
+                                                             "g3.16xlarge",]
+
+    Other instances will default to Proprietary Nvidia Driver DLAMI
+    """
+
+    base_proprietary_dlami_instances = [
```

Comment:
do we need to hard code the actual instance type, or is it fair to assume a certain instance type based on family (i.e. p3, g3, instead of all of the specific types)

Reply:
We can, I just thought it wouldn't make much of a difference between maintaining a list of family types vs list all all instance names so I opted for the more verbose one in the event of an unlikely edge case such as removing a particular instance type from proprietary dlami.

Reply:
How do you plan to keep this list up to date? Is there a reference link you can use (i.e. instead of listening everything out in the function description)?

Reply:
Added a link for all instance types https://aws.amazon.com/ec2/instance-types/#Accelerated_Computing

Reply:
The current source of truth we have publicly is here https://docs.aws.amazon.com/dlami/latest/devguide/important-changes.html

Jeet has been maintaining this page for most up to date info.